### PR TITLE
21229-Enhance-file-attribute-tests

### PR DESCRIPTION
**Pull request by akgrant43**
21229-Enhance-file-attribute-tests

Add FileSystem-Tests-Attributes package

### CODE REVIEW

**Comment by akgrant43** [comment]
Fogbugz: https://pharo.fogbugz.com/f/cases/21229/

**Comment by akgrant43** [comment]
The 12 failing tests appear to have nothing to do with the PR (which doesn't add or change any functionality, just adds tests (which I assumed passed since there isn't any error report for them)).

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+FileReferenceAttributeTests >> testIsDirectory [
+
+	self assert: self tempFileResource file isDirectory equals: false.
+	self assert: FileLocator temp isDirectory.
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testIsFile [
+
+	self assert: self tempFileResource file isFile equals: true.
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self assert: self tempFileResource file isFile.

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	creationTime := resource file creationTime.
+	self assert: resource beforeCreationTime <= creationTime.
+	self assert: resource afterCreationTime >= creationTime.
+
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testExists [
+
+	self assert: (self tempFileResource file exists).
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self assert: self tempFileResource file exists.

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+FileReferenceAttributeTests >> testFileSize [
+
+	self assert: self tempFileResource file size equals: 72.
+
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testIsBlock [
+
+	self assert: self tempFileResource file isBlock equals: false.
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self deny: self tempFileResource file isBlock.

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testIsCharacter [
+
+	self assert: self tempFileResource file isCharacter equals: false.
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self deny: self tempFileResource file isCharacter.

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testIsDirectory [
+
+	self assert: self tempFileResource file isDirectory equals: false.
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self deny: self tempFileResource file isDirectory.

File: src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+FileReferenceAttributeTests >> testIsReadable [
+
+	self assert: self tempFileResource file isReadable.
+
+]
+
+{ #category : #tests }
+FileReferenceAttributeTests >> testIsSymlink [
+
+	self assert: self tempFileResource file isSymlink equals: false.
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/FileReferenceAttributeTests.class.st]: self deny: self tempFileResource file isSymlink.

File: src/FileSystem-Tests-Attributes/DiskFileAttributesTestsResources.class.st [hunk 7]
```diff
@@ -0,0 +1,55 @@
+Class {
+	#name : #DiskFileAttributesTestsResources,
```
Comment (VincentBlondeau) [inline src/FileSystem-Tests-Attributes/DiskFileAttributesTestsResources.class.st]: A small comment for this class should be nice.

**Review by VincentBlondeau — CHANGES_REQUESTED** [review]
More tests are always welcomed! Thx!

Indeed the tests are not related. But your code can be improved: please use deny: instead of assert:equals: false and assert: instead of assert:equals: true.

I commented with the replacement line.

**Review by VincentBlondeau — APPROVED** [review]
Nice!

**Comment by akgrant43** [comment]
Hi Vincent,

I was waiting to check the status of the CI before commenting, but you got in first. :-)

Thanks for the detailed feedback. 

- I should have checked the class comment.
- I like the inspector that the debugger uses with #assert:equals:, so tend to use it without thinking :-)

Thanks for approving!

Cheers,
Alistair

**Comment by akgrant43** [comment]
As before, the CI test failures don't appear to be related to this PR.

**Comment by akgrant43** [comment]
Thanks, Vincent & Marcus!

**Pull request merged**